--- /tmp/linux-sgx/linux/installer/bin/install-sgx-psw.bin.tmpl.bak	2018-01-26 12:02:12.780343255 +0100
+++ /tmp/linux-sgx/linux/installer/bin/install-sgx-psw.bin.tmpl	2018-01-26 12:02:22.364442287 +0100
@@ -30,23 +30,6 @@
 #
 #
 
-grep intel_sgx /lib/modules/$(uname -r)/modules.builtin &> /dev/null
-if [[ $? != "0" ]]; then
-    modinfo isgx &> /dev/null
-    if [[ $? != "0" ]]; then
-        modinfo intel_sgx &> /dev/null
-        if [[ $? != "0" ]]; then
-            lsmod | grep 'isgx\|intel_sgx' &> /dev/null
-            if [[ $? = "0" ]]; then
-                echo "Warning: You did not follow the document to install the driver package"
-                echo
-            else
-                echo "Please install the driver package before the PSW package"
-                exit 4
-            fi
-        fi
-    fi
-fi
 
 set -e
 
